presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -79,42 +84,50 @@ public String toString()
                 .add("filter", filter)
                 .add("limit", limit)
                 .add("aggregations", aggregations)
+                .add("useSqlSyntax", useSqlSyntax)
                 .toString();
     }
 
-    PinotQueryGeneratorContext()
+    public PinotQueryGeneratorContext(boolean usePinotSqlSyntax)
     {
-        this(new LinkedHashMap<>(), null);
+        this(new HashMap<>(), new LinkedHashSet<>(), null, usePinotSqlSyntax);
     }
 
     PinotQueryGeneratorContext(
-            LinkedHashMap<VariableReferenceExpression, Selection> selections,
-            String from)
+            Map<VariableReferenceExpression, Selection> selections,
+            LinkedHashSet<VariableReferenceExpression> outputs,
+            String from,
+            boolean useSqlSyntax)
     {
         this(
                 selections,
+                outputs,
                 Optional.ofNullable(from),
                 Optional.empty(),
                 0,
                 new LinkedHashSet<>(),
                 new LinkedHashMap<>(),
```

Comment:
It will make the fields ordering inside OrderBy clause to be not deterministic.
E.g. Presto query `SELECT count(*), A, B FROM myTable ORDER BY A,B` will make generatedPinot query to be  either `SELECT count(*), A, B FROM myTable ORDER BY A,B` or  `SELECT count(*), A, B FROM myTable ORDER BY B, A`.
I somehow feel it's better to keep the ordering as it is in case the underlying connector may take advantage on this(which is not there yet).

Reply:
Oh okay. This is for Order-By !. I thought this is still for one of selections or outputs. Lets retain the linkedhashmap for order-by !.

Reply:
got it, will keep it :)